### DISCUSSION TITLE
How to set defualt value when using AsyncReactSelect

### DISCUSSION THREAD

**Discussion by gustavomm19** [JedWatson/react-select#5755] [Q&A]
I have the following code, which is a dropdown to filter the data on a table, so the value of the option selected is used as a query string to filter the data when the page is reloaded, keeping track of the filters applied. My problem is that when the page is reloaded I cant use the querystring to pre-select the option with the same value. `query.student` is equal to the `value` property of the selected option, but it doesnt display the selected option until I do it manually
https://codesandbox.io/s/intelligent-cherry-vkq5y?file=/src/App.js

```
<AsyncReactSelect
                id="student-select"
                placeholder={t('filter.student')}
                isClearable
                value={studentLabel}
                defaultValue={query.student}
                getOptionValue={(option) => option.value}
                height="50px"
                fontSize="15px"
                onChange={(selected) => {
                  setStudentLabel(
                    selected !== null
                      ? {
                        id: selected?.id,
                        value: selected?.value,
                        label: selected?.label,
                      }
                      : null,
                  );
                }}
                defaultOptions
                cacheOptions
                loadOptions={(inputValue) => handlers
                  .getStudents(selectedCohort?.slug || cohortSlug, academy, inputValue !== '' ? { like: inputValue } : {})
                  .then((students) => students.map((student) => ({
                    id: student.user.id,
                    value: student.user.id,
                    label: `${student.user.first_name} ${student.user.last_name}`,
                  })))}
              />
```

**Comment by manjushsh** [comment]
Hi @gustavomm19 can you please share a sample codesandbox? Thanks.

**Reply by gustavomm19** [reply]
Here it is https://codesandbox.io/s/intelligent-cherry-vkq5y?file=/src/App.js

As you can see, the `defaultStudent` is equal to 1, so I need to preselect the option with that same `value` or `id`

**Reply by manjushsh** [reply]
Hi @gustavomm19, We can achieve this by waiting for options to load then update a state as default value. Then we can pass that state value as default option to Async select

Updated [Code Sandbox link](https://codesandbox.io/s/adoring-farrell-7v88hx?file=/src/App.js)


[reference](https://codesandbox.io/s/react-select-async-default-option-value-clsfe?file=/index.js)

**Reply by nadjiel** [reply]
@manjushsh this fix works in this example because we know all the **options anticipatedly**, but I think in most cases the **options** would be **gotten from an API** and, in that context, it's **possible** the `defaultValue` (in this case `defaultStudent`) **isn't necessarily included in the options** returned (i.e.: because of pagination or something similar).

With this in mind, I think it would be greatly **appreciated** if we could have some sort of `loadDefaultOption` native to the `AsyncReactSelect` component. It would even **help avoiding some boilerplate** with `useEffect` and manually managing state.
